api/v1/views/cities.py

```diff
@@ -1 +1,77 @@
 #!/usr/bin/python3
+"""
+New view for City objects that handles default Restful API actions
+"""
+from flask import Flask, jsonify, abort, request
+from api.v1.views import app_views
+from models import storage
+from models.city import City
+from models.state import State
+
+
+@app_views.route('/api/v1/states/<state_id>/cities', strict_slashes=False)
+def all_cities(state_id):
+    """ Retrieve a list of all City objects of State """
+    all_cities = []
+    if not storage.get('State', state_id):
+        abort(404)
+    for city in storage.all('City').values():
+        if state_id == city.to_dict()['state_id']:
+            all_states.append(state.to_dict())
+    return jsonify(all_cities)
+
+
+@app_views.route('/api/v1/cities/<city_id>', strict_slashes=False)
+def retrieve_city(city_id):
+    """ Retrieve a particular City """
+    city = storage.get('City', city_id)
+    if city:
+        return city.to_dict()
+    abort(404)
+
+
+@app_views.route('/api/v1/cities/<city_id>', methods=['DELETE'],
+                 strict_slashes=False)
+def delete_city(city_id):
+    """ Delete a City """
+    city = storage.get('City', city_id)
+    if city:
+        state.delete(city)
+        storage.save()
+        return {}
+    abort(404)
+
+
+@app_views.route('/api/v1/states/<state_id>/cities', methods=['POST'],
+                 strict_slashes=False)
+def create_city(state_id):
+    """ Create a City """
+    city_name_name = request.get_json()
```

Comment:
i'm going to show my ignorance here, what's city_name_name?

Reply:
that would be a typo

Reply:
fixed now

Reply:
I thought I was just missing something!